### PR TITLE
Add very rudimentary backup monitoring

### PR DESCRIPTION
**Pull request by Kidswiss**
This adds a backup monitoring via jobs. It alerts if there's a job that has failed in any of the AppCat namespaces.

## Checklist

- [x] The PR has a meaningful title. It will be used to auto generate the
      changelog.
      The PR has a meaningful description that sums up the change. It will be
      linked in the changelog.
- [x] PR contains a single logical change (to build a better changelog).
- [x] Categorize the PR by adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog.
- [ ] Link this PR to related issues or PRs.

### CODE REVIEW

**Review by zugao — COMMENTED** [review]
Why don't we add this in the appcat common package?

**Comment by Kidswiss** [comment]
> Why don't we add this in the appcat common package?

Common is a library and doesn't produce any files by itself.

File: component/tests/golden/apiserver/appcat/appcat/10_appcat_backup_monitoring.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  groups:
+    - name: appcat-backup
+      rules:
+        - alert: AppCatBackupJobError
+          annotations:
+            description: The backup job {{ $labels.job_name }} in namespace {{ $labels.namespace
+              }} has failed.
+            runbook_url: https://kb.vshn.ch/app-catalog/how-tos/appcat/AppCatBackupJobError.html
+            summary: AppCat service backup failed.
+          expr: kube_job_failed{job_name=~".*backup.*", namespace=~"vshn-(postgres|redis|stsResizer)-.*"}
```
Comment (zugao) [inline component/tests/golden/apiserver/appcat/appcat/10_appcat_backup_monitoring.yaml:16]: Shouldn't it be postgresql instead of posgres? Why there's stsResizer as well?

Comment (Kidswiss) [reply component/tests/golden/apiserver/appcat/appcat/10_appcat_backup_monitoring.yaml]: As discussed in chat, I will refactor the `services.vshn` object, so it only contains actual vshn services.

File: component/component/main.jsonnet [hunk 2]
```diff
@@ -195,12 +195,48 @@ local emailSecret = kube.Secret(params.services.vshn.emailAlerting.secretName) {
   },
 };
 
+local filterName(name) = '|' + if name == 'postgres' then 'postgresql' else if name == 'emailAlerting' then '' else name;
+local backupJobRegex = std.foldl(function(prev, current) prev + filterName(current.name), common.FilterServiceByBoolean('enabled'), '');
```
Comment (TheBigLee) [inline component/component/main.jsonnet:199]: This regex currently produces strange (but probably valid) output in some cases. eg:
```
vshn-(||minio)-
```

File: component/class/defaults.yml [hunk 3]
```diff
@@ -438,8 +438,6 @@ parameters:
           smtpFromAddress: myuser@example.com
           secretNamespace: syn-appcat
           secretName: mailgun-smtp-credentials
-        stsResizer:
-          enabled: true
```
Comment (wejdross) [inline component/class/defaults.yml:442]: why is this removed here?

File: component/component/main.jsonnet (hunk 2 above)
Comment (wejdross) [reply component/component/main.jsonnet:199]: it allows empty value:
```
vshn--
```
is a match. that's bad. I know it's a corner case we wouldn't probably ever get, but I'd fix it just to sleep better
